### PR TITLE
Wrapper fix

### PR DESCRIPTION
**Pull request by vuolleko**
#### Summary:
Updated external wrapper to current dev.
Updated legacy notebook with some explanation.

### CODE REVIEW

**Comment by akangasr** [comment]
What changes are there to notebooks/ELFI_tutorial_MA2.ipynb, as these are not shown by github web interface (too large)? If the changes are not related to 'wrapper fixing', then they should be a separate commit.

File: elfi/env.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 
 logging.getLogger('tornado').setLevel(logging.WARNING)
+logging.getLogger('elfi.methods').setLevel(logging.INFO)
```
Comment (vuolleko) [inline elfi/env.py]: On second thought you're right, it's probably better to have in methods.

File: elfi/wrapper.py [hunk 2]
```diff
@@ -6,6 +6,7 @@ class Wrapper():
     """ Wraps an external command to work as a callable operation for a node.
 
         Currently only supports sequential operations (not vectorized).
+        You can force this by setting batch_size=1 in the ABC method.
```
Comment (vuolleko) [inline elfi/wrapper.py]: Rewritten. As it is, sequential operations don't seem to work without setting batch_size=1.

File: elfi/wrapper.py [hunk 3]
```diff
@@ -26,28 +27,28 @@ def __init__(self, command_template="", post=None, pre=None):
 
     @staticmethod
     def process_elfi_internals(command_template, args, kwargs):
-        """ Replace 'prng' in kwargs with a seed from the generator if present in template """
+        """ Replace 'random_state' in kwargs with a seed from the generator if present in template """
         proc_args = list()
         for a in args:
             if isinstance(a, np.ndarray):
-                if a.shape == (1,):
+                if a.shape[0] == 1:
```
Comment (vuolleko) [inline elfi/wrapper.py]: Done.

File: elfi/wrapper.py [hunk 4]
```diff
@@ -26,28 +27,28 @@ def __init__(self, command_template="", post=None, pre=None):
...
+                if isinstance(kwargs["random_state"], np.random.RandomState):
+                    kwargs["seed"] = str(kwargs["random_state"].randint(np.iinfo(np.uint32).max))
+            del kwargs["random_state"]
         return command_template, proc_args, kwargs
 
     @staticmethod
     def read_nparray(stdout):
         """ Interpret the stdout as a space-separated numpy array """
-        return np.fromstring(stdout, sep=" ")
+        return np.fromstring(stdout, sep=" ")[None, :]
```
Comment (vuolleko) [inline elfi/wrapper.py]: This adds the dim that core currently needs. Added documentation.

File: elfi/wrapper.py [hunk 5]
```diff
@@ -26,28 +27,31 @@ def __init__(self, command_template="", post=None, pre=None):
...
         proc_args = list()
         for a in args:
             if isinstance(a, np.ndarray):
-                if a.shape == (1,):
+                if a.shape == (1, 1):
                     # take single values out of array
-                    proc_args.append(a[0])
+                    proc_args.append(a.item())
                 else:
                     raise NotImplementedError("Wrapper does not yet support array arguments")
```
Comment (lintusj1) [inline elfi/wrapper.py]: I think would be important to have this in. Look for `elfi.tools.vectorize` as how to implement it. Would be great to use same code here and there via some smart way.

Comment (vuolleko) [inline elfi/wrapper.py]: I suggest we leave that for another PR. The current suffices for most users, I guess, and is important to have fixed quickly.
